xmtp_mls/src/groups/mod.rs

```diff
@@ -296,6 +299,39 @@ where
         }
     }
 
+    pub(crate) async fn post_commit(&self, conn: &mut DbConnection) -> Result<(), GroupError> {
+        let intents = self.client.store.find_group_intents(
+            conn,
+            self.group_id.clone(),
+            Some(vec![IntentState::Committed]),
+            None,
+        )?;
+
+        for intent in intents {
+            if intent.post_commit_data.is_some() {
+                let post_commit_data = intent.post_commit_data.unwrap();
+                let post_commit_action = PostCommitAction::from_bytes(post_commit_data.as_slice())?;
+                match post_commit_action {
+                    PostCommitAction::SendWelcomes(action) => {
+                        let welcomes: Vec<WelcomeMessage> = action
+                            .installation_ids
+                            .into_iter()
+                            .map(|installation_id| WelcomeMessage {
+                                installation_id,
+                                ciphertext: action.welcome_message.clone(),
+                            })
+                            .collect();
+                        self.client.api_client.publish_welcomes(welcomes).await?;
+                    }
+                }
+            }
+            let deleter: &mut dyn Delete<StoredGroupIntent, Key = i32> = conn;
```

Comment:
I had to do some gymnastics to make the Delete trait work now that we have multiple structs implementing it. Now that it requires disambiguation it's...not beautiful. Maybe there's some better way to tell the compiler which model we're deleting, or maybe we should rethink the trait to be explicitly generic.